geocodingsearchv7/client.go

```diff
@@ -152,3 +160,58 @@ func checkResponse(r *http.Response) error {
 	}
 	return &responseError{Response: &response}
 }
+
+// checkResponseXML checks the API response for errors, and returns them if present. A response is considered an
+// error if it has a status code outside the 200 range. It decodes the error as XML.
+func checkResponseXML(r *http.Response) error {
```

Comment:
Place this function below the DoXML method? Either to have public functionality above private or because it's only used by the DoXML method.